src/test_sudoku.c

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file test_sudoku.c
+ *
+ * Purpose  : This function tests whether the regenerated sudoku meets all the sudoku rules or not .
+ *
+ * Language : C
+ *
+ * Author   : Rishith Rao - rishithrao@cmail.carleton.ca
+ *            Prasanth Kota - prasanthkota@cmail.carleton.ca
+ *
+ * Date     : 13 June, 2020.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "../include/sudoku.h"
+
+struct node {
+    int data;
+    struct node *next;
+};
+
+struct node *head = NULL;
+struct node *current = NULL;
+
+/**
+ * Deletes the first element of the array.
+ */
+struct node* deleteFirst() {
+    struct node *tempLink = head;/*save reference to first link*/
+    head = head->next;/*mark next to first link as first*/
+    return tempLink;/*return the deleted link*/
+}
+
+/**
+ * Checks whether the list is empty or not.
+ */
+bool isEmpty() {
+    return head == NULL;
+}
+
+/**
+ * Insert link at the first location.
+ */
+void insertFirst(int data){
+    struct node *link = (struct node*) malloc(sizeof(struct node));/*create a link*/
+    link->data = data;
+    link->next = head;/*point it to old first node*/
+    head = link;/*point first to new first node*/
+}
+
+/**
+ * Delete a link with given data
+ */
+struct node* delete(int data) {
+    struct node* current = head;/*start from the first link*/
+    struct node* previous = NULL;
+    if(head == NULL){/*if list is empty*/
+        return NULL;
+    }
+    while(current->data != data){/*navigate through list*/
+        if(current->next == NULL) {/*if it is last node*/
+            return NULL;
+        }else {
+             previous = current;/*store reference to current link*/
+             current = current->next;/*move to next link*/
+         }
+    }
+    if(current == head){/*found a match, update the link*/
+        head = head->next;/*change first to point to next link*/
+    }else {
+        previous->next = current->next;/*bypass the current link*/
+     }    
+    return current;
+}
+
+/*
+ *This checks the 3X3 blocks for repeated values.
+ */
+_Bool test_3X3(int rearranged_sudoku[][9]) {
+    int z[9][9];  
+    int x=0,y=2,sum=0,k=0,l=2,a,sumplus=0;
+    int flag=0;
+    while(x<=8) {
+        while(k<=8) {
```

Comment:
I see that your code checks for a 3x3 block, but what about cols and rows? I don't see how this function can verify if two identical digits are not in the same row or col?

Reply:
Tasks are divided between me and Rishith. Rows/Columns will be done by me and Rishith does the rest.